routers/repo/commit.go

```diff
@@ -388,6 +388,12 @@ func RawDiff(ctx *context.Context) {
 		git.RawDiffType(ctx.Params(":ext")),
 		ctx.Resp,
 	); err != nil {
+		if git.IsErrNotExist(err) {
+			ctx.JSON(404, map[string]interface{}{
+				"message": "commit " + ctx.Params(":sha") + " does not exist.",
+			})
+			return
```

Comment:
GH just send a text reply  - what's the reasoning behind the json?

Reply:
This is not API, right? Should probably send the 404 HTML page.

Reply:
it's neither API nor html. it's a text endpoint for getting the diffs.

Reply:
Hmm GH sends "Not found" in such a case, but if I go via browser, I still get a HTML page for some reason. Does not seem to be rely on user-agent sniffing thought.

https://github.com/go-gitea/gitea/pull/9999999.patch

Reply:
Ah, they do it via the `accept` header. This gives HTML:

```
curl -L -H 'accept: text/html' https://github.com/go-gitea/gitea/pull/9999999.patch
```

Reply:
Hmm, so It's okay to response html page.  and we can change ``ctx.NotFound`` ro response different message type for different ``content-type`` request.